x/mongo/driver/topology/connection.go

```diff
@@ -292,10 +294,16 @@ func (c *connection) readWireMessage(ctx context.Context, dst []byte) ([]byte, e
 	if err != nil {
 		// We closeConnection the connection because we don't know if there are other bytes left to read.
 		c.close()
+		var message string
```

Comment:
nit: We generally try to avoid if/else statements. You can initialize `message` to a default and then overwrite it in a condition. The general case is "incomplete read of message header" so this would look like:

```
message := "incomplete read of message header"
if err == io.EOF {
    message = "Socket was unexpectedly closed"
}
```

Reply:
done